### ISSUE TITLE
[Bug]: Deployments don't start anymore

### ISSUE THREAD

**Issue by JensUweB** [coollabsio/coolify#4542]
### Error Message and Logs

Our Coolify deployment consists of 4 servers total. The Coolify admin host, a build server & two hosts to deploy on with two separate subdomains. Everything worked fine until a few days ago.

When we try to deploy an app (with custom Dockerfile or Nixpacks, with or without build server - doesn't matter) the deployment gets added to the queue and sits there forever. Pressing the "force start" changes the state to "processing", but nothing happens either. No messages, no logs in the build window, nothing. Sometimes restarting the whole admin host gets it working again, but only for a short amount of time. But this "workaround" seems to have stopped working for us now too leaving us unable to deploy anything.

As soon as I press cancel a message pops up with an error:

> instant_remote_process(): Argument #2 ($server) must be of type App\Models\Server, null given, called in /var/www/html/app/Livewire/Project/Application/DeploymentNavbar.php on line 80

The proxy has logged some network errors this morning, but they are probably related to #4483 - an Issue we also had a couple of times.

> 2024-12-11T09:49:58.385076078Z 2024-12-11T09:49:58Z ERR error="accept tcp [::]:80: use of closed network connection" entryPointName=http
2024-12-11T09:49:58.385157225Z 2024-12-11T09:49:58Z ERR Error while starting server error="accept tcp [::]:80: use of closed network connection" entryPointName=http
2024-12-11T09:49:58.385163785Z 2024-12-11T09:49:58Z ERR error="accept tcp [::]:443: use of closed network connection" entryPointName=https
2024-12-11T09:49:58.385169456Z 2024-12-11T09:49:58Z ERR Error while starting server error="accept tcp [::]:443: use of closed network connection" entryPointName=https
2024-12-11T09:50:08.381943611Z 2024-12-11T09:50:08Z ERR Failed to list containers for docker error="Get \"http://%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/json\": context canceled" providerName=docker
2024-12-11T10:08:28.719311029Z 2024-12-11T10:08:28Z ERR error="accept tcp [::]:80: use of closed network connection" entryPointName=http
2024-12-11T10:08:28.721598935Z 2024-12-11T10:08:28Z ERR error="accept tcp [::]:443: use of closed network connection" entryPointName=https
2024-12-11T10:08:28.725836095Z 2024-12-11T10:08:28Z ERR error="close tcp [::]:80: use of closed network connection" entryPointName=http
2024-12-11T10:08:28.760504931Z 2024-12-11T10:08:28Z ERR error="close tcp [::]:443: use of closed network connection" entryPointName=https

### Steps to Reproduce

I'm not sure how to reproduce this. I just updated my private Coolify instance to v4.0.0-beta.376 to compare it with our business instance, but deployments there still work (only 1 host & no build server).

### Coolify Version

v4.0.0-beta.376

### Are you using Coolify Cloud?

No (self-hosted)

### Operating System and Version (self-hosted)

Ubuntu 22.04

### Additional Information

We use shared servers from Hetzner Cloud for Coolify.

Edit:
Our Shard Host 1 was at 100% cpu usage. After I restarted it usage went back down to like 50%. After that deployments in Coolify started working again. But why should the Admin Panel & deployments on Host 2 be affected by the cpu usage at host 1? Maybe just a coincidence.

**Comment by ludioao** [comment]
Same issue here.

**Comment by mindhaq** [comment]
I have a similar issue with current Coolify cloud.

**Comment by khoi-fish** [comment]
+1 one to this. Just started happening to me today

**Comment by emimontesdeoca** [comment]
+1 same to me

**Comment by NathanGrenier** [comment]
+1 same here

**Comment by enginesa** [comment]
+1

**Comment by ricky-cc** [comment]
+1

**Comment by peaklabs-dev** [comment]
@JensUweB The issues could be if host 1 was at 100% cpu usage and that is the server where Coolify runs on, right? because then it could be that the jobs getting done on horizon on the Coolify server were not executing properly and communicating with your other servers so you deployments are stuck.
It would really help if you could monitor resource usage and when deployments keep failing and report back to this issue, as this could be the cause.

**Comment by JensUweB** [comment]
> [@JensUweB](https://github.com/JensUweB) The issues could be if host 1 was at 100% cpu usage and that is the server where Coolify runs on, right? because then it could be that the jobs getting done on horizon on the Coolify server were not executing properly and communicating with your other servers so you deployments are stuck. It would really help if you could monitor resource usage and when deployments keep failing and report back to this issue, as this could be the cause.

We have 3 servers in total. 1 for the admin panel and two where we host our applications on.
Last time this error occurred the cpu usage of the admin panel host was fine, but one of the others was at 100% CPU usage and wasn't even responding to a gracefull shutdown call.

I'm not sure if Coolify or one of our apps causes this cpu spike. We're running multiple Strapi instances there (besides other Node.js based stuff) and I have the suspicion that one of the Strapi's was going parabolic (I don't have resource limits on everything - which maybe was a mistake :D ).

Whats about the others that have similar issues? Do you guys can add some details too?

**Comment by soranoo** [comment]
Same issue here, anyone figure it out?

**Comment by michalnag** [comment]
I am facing the same problem. It started with noticing CPU being permanently at 100% usage. After some digging around I noticed that it runs multiple `[horizon](https://laravel.com/docs/11.x/horizon)` services using `supervisord`. As a very nasty hack, I logged in to `coolify` container, reduced config for `maxProcesses` to `2`, restarted container and at least CPU is at manageable levels. I still haven't figured out why deployments don't work (even if forced) :/ I will definitely update this thread once I find anything but I am leaning towards setting up new fresh instance as this happened to me after setting up old legacy project that failed multiple times. It makes me wonder if there are some different queue cleaners other than `laravel` ones that can be used inside container with `artisan`....

**Comment by astrooom** [comment]
I'm having the same issue - however the deploment **does** start. It just takes from 1-10 minutes. No Crazy CPU or RAM usage at all.

**Comment by Nexulo** [comment]
Yes, this really is a major problem and has been going on for what feels like an eternity.

**Comment by namgo** [comment]
I've experienced what I believe to be the same issue consistently for the past (3?) months, in that the deployment process is incredibly slow to start but does start eventually.

I have noticed almost no cpu usage on either the coolify host or the server it's deploying to, no increase in memory consumption either. For all I can tell literally nothing is happening on either side.

On my end the deployment does start after ~15-20 minutes, and confusingly there appears to **usually** be instant deployment if I deploy a single host rather than multiple hosts.

For some added context, I queue deployments all at once rather than off of git pushes, the deployments show up in queue and stay as "queued" without any activity.

There are no logs indicating a failure on either end - although if there's some kind of verbosity level I can adjust please let me know.

**Comment by namgo** [comment]
Is it possible that Slack notification queues are causing problems? After weeks of incredibly slow-to-queue deployments we disabled our slack integration and our deployments appear to be back to normal for us.

**Comment by oussamadouhou** [comment]
Is there any information i can share more than this as I face the same issue:

Coolify portal:
OS: Linux coolify 6.8.12-15-pve #1 SMP PREEMPT_DYNAMIC PMX 6.8.12-15 (2025-09-12T11:02Z) x86_64 GNU/Linux
Coolify: v4.0.0-beta.434

Deployment server:
Linux 01 6.1.0-32-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.129-1 (2025-03-06) x86_64 GNU/Linux
CPU: 15% RAM: 90% available HDD: 30% used

**Comment by ReinMengelberg** [comment]
`docker restart coolify` after ssh'ing into the server as root fixed it for me.

**Comment by rootz491** [comment]
I'm facing the same issue.

**Comment by andrasbacsai** [comment]
Closing old issue (which is probably fixed already). Reopen if still reproducible.
